aptos-move/aptos-aggregator/src/aggregator_change_set.rs

```diff
@@ -0,0 +1,507 @@
+use move_core_types::vm_status::{StatusCode, VMStatus};
+
+/// When merging aggregator changes of two transactions,
+/// unable to merge the histories of the transactions.
+pub(crate) const EMERGE_HISTORIES: u64 = 0x02_0008;
+
+/// Represents a single aggregator change.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct AggregatorChange {
+    pub max_value: u128,
+    pub state: AggregatorState,
+    // `base_aggregator` is None for Aggregators.
+    // `base_aggregator` is Some(id) for AggregatorSnapshots.
+    pub base_aggregator: Option<AggregatorID>,
+}
+
+impl AggregatorChange {
+    /// TODO: Implement this.
+    pub fn materialize_aggregator(
+        self,
+        _state_view: &dyn StateView,
+        _aggregator_id: &AggregatorID,
+    ) -> anyhow::Result<AggregatorChange, VMStatus> {
+        Ok(self)
+    }
+
+    /// Returns the result of applying the AggregatorState to the base value
+    /// Returns error if postcondition is not satisfied.
+    pub fn apply_aggregator_change_to(&self, base: u128) -> PartialVMResult<u128> {
+        match self.state {
+            AggregatorState::Data { value } => Ok(value),
+            AggregatorState::Delta { delta, .. } => {
+                // First, validate if the current delta operation can be applied to the base.
+                validate_history(base, self.max_value, &self.state)?;
+                Ok(addition_deltavalue(base, delta, self.max_value)?)
+            },
+        }
+    }
+
+    /// Applies self on top of previous delta, merging them together. Note
+    /// that the strict ordering here is crucial for catching overflows
+    /// correctly.
+    ///
+    /// TODO: What happens if the base aggregator is not none?
+    pub fn merge_with_previous_aggregator_change(
+        &mut self,
+        previous_change: AggregatorChange,
+    ) -> PartialVMResult<()> {
+        assert_eq!(
+            self.max_value, previous_change.max_value,
+            "Cannot merge aggregator changes with different max_values",
+        );
+        // When the previous aggregator change is a snapshot, we cannot merge it with the current.
+        if previous_change.base_aggregator.is_some() {
+            return Err(
+                PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
+                    .with_message("Cannot merge aggregator snapshots".to_string()),
+            );
+        }
+        match self.state {
+            AggregatorState::Data { .. } => {
+                // If the current state is Data, then merging with previous state won't change anything.
+                return Ok(());
+            },
+            AggregatorState::Delta { delta, history, .. } => {
+                match previous_change.state {
+                    AggregatorState::Data { value } => {
+                        // When prev_state is Data { value }, and current state is Delta { delta }, merging them into Data { value + delta }
```

Comment:
nit: some comment lines seem to short, while some others seem too long (like this one)